packages/frontend/src/features/sqlRunner/components/SaveCustomViewModal.tsx

```diff
@@ -0,0 +1,245 @@
+import {
+    CustomViewType,
+    DbtProjectType,
+    getProjectDirectory,
+    snakeCaseName,
+} from '@lightdash/common';
+import {
+    Alert,
+    Badge,
+    Button,
+    Group,
+    List,
+    Modal,
+    Radio,
+    Stack,
+    Text,
+    TextInput,
+    Tooltip,
+    type ModalProps,
+} from '@mantine/core';
+import { useForm, zodResolver } from '@mantine/form';
+import { IconInfoCircle, IconWriting } from '@tabler/icons-react';
+import { useCallback, useMemo, type FC } from 'react';
+import { z } from 'zod';
+import MantineIcon from '../../../components/common/MantineIcon';
+import { useProject } from '../../../hooks/useProject';
+import { useCreateCustomExplore } from '../hooks/useCustomExplore';
+import { useAppSelector } from '../store/hooks';
+
+const getCustomViewTypeAlertText = (customViewType: CustomViewType) => {
+    switch (customViewType) {
+        case CustomViewType.VIRTUAL:
+            return "Create a virtual custom view so others can reuse this query in Lightdash, but it won't be written to or managed in your dbt project.";
+        case CustomViewType.PERSISTENT:
+            return 'Create a new model in your dbt project from this SQL query. This will create a new branch and start a pull request (docs coming soon).';
+        default:
+            return undefined;
+    }
+};
+
+const validationSchema = z.object({
+    name: z.string().min(1),
+    customViewType: z.nativeEnum(CustomViewType),
+});
+
+type FormValues = z.infer<typeof validationSchema>;
+
+type Props = ModalProps;
+
+export const SaveCustomViewModal: FC<Props> = ({ opened, onClose }) => {
+    const projectUuid = useAppSelector((state) => state.sqlRunner.projectUuid);
+    const sql = useAppSelector((state) => state.sqlRunner.sql);
+    const columns = useAppSelector((state) => state.sqlRunner.sqlColumns);
+    const { mutateAsync: createCustomExplore, isLoading } =
+        useCreateCustomExplore({
+            projectUuid,
+        });
+    const form = useForm<FormValues>({
+        initialValues: {
+            name: '',
+            customViewType: CustomViewType.VIRTUAL,
+        },
+        validate: zodResolver(validationSchema),
+    });
+
+    const { data: project } = useProject(projectUuid);
+    const canWriteToDbtProject = !!(
+        project?.dbtConnection.type === DbtProjectType.GITHUB
+    );
+    // TODO: Check if possible to create a branch and PR - if not, and it's not a git project, disable option with tooltip
```

Comment:
First you can check `feature flag`, then `    "hasGithub": true,` on health, then you can check if the user has logged in in github with `/api/v1/github/repos/list`

Reply:
What feature flag do you want me to check here @rephus ?

On the login-check, I've created a separate ticket: https://github.com/lightdash/lightdash/issues/11742